### DISCUSSION TITLE
zh/help/

### DISCUSSION THREAD

**Discussion by giscus** [ultralytics/ultralytics#19271] [Docs]
# zh/help/

请访问Ultralytics 帮助中心，这里有各种指南、常见问题、CI 流程和政策，可为您的YOLO 模型体验和贡献提供支持。

https://docs.ultralytics.com/zh/help/

**Comment by awgawgcvawgwg61681** [comment]
Hello, I need some help. I've developed a desktop tool where I load a Vue interface I wrote using pywebview, and access the interfaces in Python through the HTTP protocol. These interfaces are for initializing models and performing inference with these models. There are two models: one is a behavior classification model trained with YOLOv8, and the other is yolo11n-pose.pt. Both models are loaded during initialization. However, when I package the application with PyInstaller and run it on another computer, the models take a very long time to load, as if they are downloading something. But I am quite sure that the access path in my software indeed points to the local model files, and it runs successfully on my own computer.

**Reply by glenn-jocher** [reply]
@awgawgcvawgwg61681 thank you for reaching out! For PyInstaller packaging issues with YOLO models, please ensure your spec file explicitly bundles the model files using `datas=[('yolo11n-pose.pt', '.')]` and verify local file paths with absolute paths in your code. The models shouldn't require downloads if properly included. For additional packaging guidance, see our PyInstaller example in the [Ultralytics HUB Docs](https://docs.ultralytics.com/hub/). Let us know if you need more specific details! 🚀

**Comment by UltralyticsAssistant** [comment]
👋 Hello, thank you for your interest in Ultralytics 🚀! It seems like you're referring to the [Chinese Help Center](https://docs.ultralytics.com/zh/help/) for Ultralytics and YOLO. This is a great resource for guidance, FAQs, and support for your YOLO model usage or contributions.

If this is a 🐛 Bug Report, we kindly request you to provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help our team debug it efficiently.

For custom training ❓ Questions or other specific issues, please provide detailed information, including code snippets, dataset examples, error messages, and training logs. To ensure your model performs at its best, check out our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

Here are a few quick steps you can follow to ensure your environment is up-to-date:  
1. Upgrade the `ultralytics` package and dependencies in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/):

```bash
pip install -U ultralytics
```

2. Ensure you're using one of our verified environments:

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM: [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI: [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**: [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

💬 Additionally, feel free to join the Ultralytics community where you can connect with others:
- Real-time chat on [Discord](https://discord.com/invite/ultralytics) 🎧
- In-depth discussions on [Discourse](https://community.ultralytics.com/)
- Explore and share on our [Subreddit](https://reddit.com/r/Ultralytics)

Lastly, if all our [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are passing and the status badge is green ✅, it ensures correct operation across all platforms (macOS, Windows, Ubuntu) every 24 hours and on every commit. You can find the CI badge here: <a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>.

This is an automated response 🤖 to ensure you get started quickly. An Ultralytics engineer will review and assist with your discussion soon!

**Comment by strawberries008** [comment]
Hello, I would like to ask if you do not need to download the pre-trained model separately after installing the latest Ultralytics library, because I found that you do not need to download the pre-trained model, but only need to specify the command line to train, and whether the Ultralytics library includes them. I look forward to your reply. Thank you.

**Reply by glenn-jocher** [reply]
@strawberries008 thanks for your question! Yes, Ultralytics YOLO11 models are included in the library and download automatically when first used. For details on available models, see the Ultralytics YOLO11 Models documentation.

**Reply by strawberries008** [reply]
Thank you for your reply! Wish you a happy life.

**Reply by glenn-jocher** [reply]
@strawberries008 thank you for your kind words! We're glad our tools are helpful for you. The Ultralytics team appreciates your support and wishes you continued success with your computer vision projects! 🚀

**Comment by diamondeyess** [comment]
When analyzing the detection task of YOLO11, I have two questions regarding the tflite format (with INT8 input and output):

For the INT8 output in the format of [batch, features, bbox], how can I determine the specific composition and order of the "features"?
After obtaining the output and performing dequantization, is it still necessary to normalize the results? (The reason is that after parsing the dequantization results, I found that some of the classification scores are greater than 1. I assume that the "features" consist of x, y, w, h, and all classification scores.)

**Comment by LJY886321** [comment]
onnx输出的[1,6,8400],中的6是包含一个类别数吗，为什么我的pt文件是两个类别，而onnx总是一个类别数呢

**Reply by glenn-jocher** [reply]
`[1, 6, 8400]` 里这个 `6` 通常不是“只有 1 个类别”，而是 `4 + nc`：前 4 个通道是框坐标，后 `nc` 个通道是每个类别的分数；所以当 `6=4+2` 时，恰好表示你的模型是 **2 类**（与你的 `pt` 一致），只是导出后的张量结构里**没有单独的 objectness 通道**（端到端/NMS-free 头一般用 per-class score 形式）。相关输出说明见[导出模型的输出张量说明](https://docs.ultralytics.com/modes/export/)。如果你愿意，把你的导出命令（是否带了 `nms=True`）和 ONNX 的输出节点名贴在这里，我可以帮你对齐后处理的维度/转置方式。

**Comment by 995742853** [comment]
Hi. I noticed in ultralytics/utils/torch_utils.py, the initialize_weights function sets eps=0.001 and momentum=0.03 for BatchNorm2d layers, instead of the PyTorch defaults.

Could you briefly explain the reasoning behind these values? Do they significantly improve model performance or robustness?

Thanks!

**Reply by glenn-jocher** [reply]
Those BatchNorm2d settings (`eps=1e-3`, `momentum=0.03`) are a long-standing YOLO training convention rather than a random tweak: the slightly larger `eps` is mainly for numerical stability (especially with mixed precision and aggressive augmentations), and the lower `momentum` makes the running mean/var update more smoothly (less noisy), which can help robustness when batch composition varies. In practice they rarely “unlock” big mAP gains by themselves, but they tend to reduce edge-case instability and keep training behavior consistent across YOLO models; see the background in our [Batch Normalization overview](https://www.ultralytics.com/glossary/batch-normalization).

**Reply by 995742853** [reply]
Thank you for your reply. Wish you a wonderful day!

**Reply by glenn-jocher** [reply]
Glad it helped—if any follow-up questions come up, please drop them here in the discussion so the Ultralytics team and community can weigh in.
